ydb/core/tx/datashard/prefix_kmeans.cpp

```diff
@@ -409,82 +408,59 @@ class TPrefixKMeansScan final: public TPrefixKMeansScanBase, private TCalculatio
         // Exact seek with Lower also possible but needs to rewrite some code in Feed
         Lead.To(Prefix.GetCells(), NTable::ESeek::Upper);
         Prefix = {};
+        IsFirstPrefixFeed = true;
+        IsPrefixRowsValid = true;
+        PrefixRows.Clear();
         MaxProbability = std::numeric_limits<ui64>::max();
         MaxRows.clear();
         Clusters.clear();
         ClusterSizes.clear();
-        HasNextPrefix = false;
         AggregatedClusters.clear();
-        return true;
     }
 
 public:
-    TPrefixKMeansScan(const TUserTable& table, TLead&& lead, NKikimrTxDataShard::TEvPrefixKMeansRequest& request,
+    TPrefixKMeansScan(const TUserTable& table, NKikimrTxDataShard::TEvPrefixKMeansRequest& request,
                       const TActorId& responseActorId, TAutoPtr<TEvDataShard::TEvPrefixKMeansResponse>&& response)
-        : TPrefixKMeansScanBase{table, std::move(lead), request, responseActorId, std::move(response)}
+        : TPrefixKMeansScanBase{table, request, responseActorId, std::move(response)}
     {
         this->Dimensions = request.GetSettings().vector_dimension();
-        LOG_D("Create " << Debug());
+        LOG_I("Create " << Debug());
     }
 
     EScan Seek(TLead& lead, ui64 seq) final
     {
-        LOG_D("Seek " << Debug());
-        ui64 zeroSeq = 0;
-        while (true) {
-            if (State == UploadState) {
-                if (MoveToNextPrefix()) {
-                    zeroSeq = seq;
-                    continue;
-                }
-                
-                if (UploadInProgress()
-                    || TryUpload(LevelBuf, LevelTable, LevelTypes, false)
-                    || TryUpload(PostingBuf, PostingTable, PostingTypes, false)
-                    || TryUpload(PrefixBuf, PrefixTable, PrefixTypes, false))
-                {
-                    return EScan::Sleep;
-                }
-
-                return EScan::Final;
+        LOG_D("Seek " << seq << " " << Debug());
+
+        if (IsExhausted) {
+            if (UploadInProgress()
```

Comment:
`UploadInProgress` already checked inside `TryUpload` (IMO should be assert inside `TryUpload`, to make it more easy to reader)

Reply:
Just an optimization (no need to pass TryUpload arguments) and easier to read code 

Also somewhere it is used in another UploadInProgress / TryUpload combination